### PR TITLE
Add cmake support

### PR DESCRIPTION
**Pull request by fabiencastan**
Hi,

This PR allows building the openFX repository with cmake.
It also simplifies the creation of openFX plugins with cmake.

You can see an example of plugin based on this PR:
https://github.com/tuttleofx/ofxPluginTemplate/blob/master/CMakeLists.txt

### CODE REVIEW

**Comment by fabiencastan** [comment]
We think that this PR will help people to create OpenFX plugins more easily.
Could we have your feedback?

File: cmake/OpenFX.cmake [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        set_target_properties(${PLUGIN_NAME} PROPERTIES LINK_FLAGS "-fvisibility=hidden,--version-script=${OFX_SUPPORT_HEADER_DIR}/linuxSymbols")
+    else()
+        set_target_properties(${PLUGIN_NAME} PROPERTIES LINK_FLAGS "-Wl,-fvisibility=hidden,--version-script=${OFX_SUPPORT_HEADER_DIR}/linuxSymbols")
+        set_target_properties(${PLUGIN_NAME} PROPERTIES INSTALL_RPATH "$ORIGIN/../../Libraries")
+    endif()
+
+    # Install OFX plugin
+    get_ofx_architecture(OFX_ARCH)
+    install(TARGETS ${PLUGIN_NAME}
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/OFX/${PLUGIN_NAME}.ofx.bundle/Contents/${OFX_ARCH}"
```
Comment (ferdnyc) [inline cmake/OpenFX.cmake]: That looks suspiciously like a path that will only work on MacOS.

File: Support/Plugins/include/ofxsProcessing.H [hunk 2]
```diff
@@ -134,10 +134,10 @@ namespace OFX {
             preProcess();
 
             // make sure there are at least 4096 pixels per CPU and at least 1 line par CPU
-            unsigned int nCPUs = (std::min(_renderWindow.x2 - _renderWindow.x1, 4096) *
+            unsigned int nCPUs = ((std::min)(_renderWindow.x2 - _renderWindow.x1, 4096) *
```
[the remark this answers is not in the document]
Comment (ferdnyc) [reply Support/Plugins/include/ofxsProcessing.H]: It's possible updating to at least `cmake_minimum_version(3.1)` and using `set(CMAKE_CXX_STANDARD 11)` to compile with `-std=c++11` (or higher) would fix this.

File: Support/Plugins/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+include(OpenFX)
+
+create_ofx_plugin("Basic")
+create_ofx_plugin("Field")
+create_ofx_plugin("Generator")
+create_ofx_plugin("Invert")
+create_ofx_plugin("MultiBundle")
+create_ofx_plugin("Retimer")
+create_ofx_plugin("Tester")
+create_ofx_plugin("Transition")
```
Comment (ferdnyc) [inline Support/Plugins/CMakeLists.txt]: For simplicity I'd do this as
```cmake
set(PLUGIN_LIST
    Basic
    Field
    Generator
    Invert
    MultiBundle
    Retimer
    Tester
    Transition
)

foreach(N IN LISTS PLUGIN_LIST)
    create_ofx_plugin($N)
endforeach()
```
Then you can just edit the list to add/remove plugins.

File: cmake/OpenFX.cmake [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        message(STATUS "Plugin ${PLUGIN_NAME}: OpenGL not found. Please set OPENGL_LIBRARIES.")
+    endif()
+    target_link_libraries(${PLUGIN_NAME} ${OPENGL_LIBRARIES})
+
+    # Add extra flags to the link step of the plugin
+    if(APPLE)
+        set_target_properties(${PLUGIN_NAME} PROPERTIES LINK_FLAGS "-bundle -fvisibility=hidden -exported_symbols_list,${OFX_SUPPORT_HEADER_DIR}/osxSymbols")
+        set_target_properties(${PLUGIN_NAME} PROPERTIES INSTALL_RPATH "@loader_path/../Frameworks;@loader_path/../Libraries")
+    elseif(WIN32)
+        set_target_properties(${PLUGIN_NAME} PROPERTIES LINK_FLAGS "-fvisibility=hidden,--version-script=${OFX_SUPPORT_HEADER_DIR}/linuxSymbols")
```
Comment (ferdnyc) [inline cmake/OpenFX.cmake]: `linuxSymbols`? On Win32? I find it difficult to imagine this working with MSVC. MinGW, maybe. (But, I could be wrong.)

**Comment by fabiencastan** [comment]
@ferdnyc
Hello, Just discovering your remarks.
I'm not involved on that topic anymore, but feel free to adjust as you want.
It would still make sense to have cmake support in ofx.

**Comment by garyo** [comment]
Hi folks; now that we are joining the ASWF and ramping up OpenFX participation, we would like to address older PRs like this. Sorry this did not get sufficient attention at the time. Is either of you interested in pursuing this? It looks like it would be a useful addition, and it's probably pretty close to working.

**Comment by ferdnyc** [comment]
@garyo I'm not interested in pursuing this myself, no. Not beyond offering what review input I'm able to contribute.

**Comment by ferdnyc** [comment]
@garyo I haven't tested it at all, but based on a quick read-through PR #92 from @Iswennz looks pretty good to me, and appears to contain everything provided in this PR, and more.

**Comment by garyo** [comment]
Closing in favor of #92 .

**Pull request closed without merging**